project/src/main/webapp/script.js

```diff
@@ -20,17 +20,161 @@ window.onload = function() {
+          `<tr>
+             <th>Candidate Name</th>
+             <th>Party Affiliation</th>
+             <th>Incumbent?</th>
+           </tr>`;
+      for (let candidateIndex = 0; candidateIndex < candidates.length;
+          candidateIndex++) {
+        const candidate = candidates[candidateIndex];
+        // Add candidate info and embed candidate ID into the URL.
+        candidatesTable.innerHTML +=
+            `<tr>
+               <td><a href="candidate.html?candidateId=${candidate.id}">
+                   ${candidate.name}</a></td>
+               <td>${candidate.partyAffiliation}</td>
+               <td>${candidate.incumbency}</td>
+             </tr>`;
+      }
+      positionItem.appendChild(candidatesTable);
+      positionsList.appendChild(positionItem);
+    }
+    positionsElement.appendChild(positionsList);
+    electionElement.appendChild(positionsElement);
+    electionsContainer.appendChild(electionElement);
+  }
+}
+
+/**
+ * Adds candidate information to the candidate page.
+ */
+async function addCandidateInformation() {
+  const candidateId = location.search.substring(
+      location.search.indexOf('=') + 1);
+
+  // Send GET request to /candidate and fetch JSON formatted data for the given
+  // candidate ID.
+  const response = await fetch(`/candidate?candidateId=${candidateId}`);
+  const dataPackage = await response.json();
 
   // Unpack response.
+  const officialCandidateInfo = dataPackage.candidateData;
+  const newsArticles = dataPackage.newsArticlesData;
+  const socialMedia = dataPackage.socialMediaData;
 
-  // Add official election/candidate information to HTML.
+  // Add 3 components of information to the candidate page.
+  addOfficialCandidateInformation(officialCandidateInfo);
+  addNewsArticles(newsArticles);
+  addSocialMedia(socialMedia);
+}
 
-  // Add URLs, that send GET request to /candidate with candidate ID, to HTML.
+/**
+ * @TODO [Adds the official information component to the candidate page.]
+ */
+function addOfficialCandidateInformation() {}
 
+/**
+ * Adds the news articles component to the candidate page.
+ *
+ * Each piece of news article has the following HTML DOM structure:
+ *     <article id="news-article-i">
+ *       <h3>News article title</h3>
+ *       <a href="URL">Source</a>
+ *       <h5>Publisher</h5>
+ *       <time>Published date</time>
+ *       <div class="news-article-content">
+ *         <p>Content [first 100 words]</p>
+ *         <a href="URL">Read the full article</a>
+ *       </div>
+ *       <hr>
+ *     </article>
+ */
+function addNewsArticles(newsArticles) {
+  const newsArticlesContainer = document.getElementById('news-articles-container');
+  newsArticlesContainer.innerHTML = '';
+  for (let i = 0; i < newsArticles.length; i++) {
+    const newsArticle = newsArticles[i];
+    const articleElement = document.createElement('article');
+    articleElement.id = `news-article-${i + 1}`;
+    articleElement.innerHTML =
+        `<h3>${newsArticle.title}</h3>
```

Comment:
Maybe you can create a more generic function that replicates this behavior?

Reply:
Thanks for the suggestion! At the moment, I believe the construction news articles/elections/positions/candidates HTML is too irregular. A generic function would require a large number of parameters that might make the function not so generic.
However, I believe this is an important point, and I have made a note of this. I will circle back to see if extracting a generic behavior is suitable, after completing the migration to Angular JS.